pom.xml

```diff
@@ -514,7 +514,7 @@
         </plugin>
         <plugin>
           <artifactId>maven-invoker-plugin</artifactId>
-          <version>3.2.0</version>
+          <version>3.1.0</version>
```

Comment:
are you sure you intended this?

Reply:
Yes. I should have written about it.

I was getting errors when running the `mvn clean install -Prun-its -e -V -nsu` command while building the `maven-failsafe-plugin` subproject. These were NPE errors in the `InstallerTask` class for the `maven-invoker-plugin`. The errors were related to processing the `dependency-reduced-pom.xml` file.

After downgrade to previous version, project build fine.

I was building on OpenJDK 8u202 with Maven 3.6.0 on Ubuntu 18.04

Reply:
Update. Today, at work, I've run a same build with invoker `3.2.0` and it installed the file just fine. The error is when you restore build at a point. Like this: `mvn clean install -Prun-its -e -V -nsu -rf maven-failsafe-plugin`

Full error:

```
[INFO] --- maven-shade-plugin:3.1.1:shade (default) @ maven-failsafe-plugin ---
[INFO] Excluding org.apache.maven.surefire:maven-surefire-common:jar:3.0.0-SNAPSHOT from the shaded jar.
[INFO] Excluding org.apache.maven.surefire:surefire-api:jar:3.0.0-SNAPSHOT from the shaded jar.
[INFO] Excluding org.apache.maven.surefire:surefire-logger-api:jar:3.0.0-SNAPSHOT from the shaded jar.
[INFO] Excluding org.apache.maven.surefire:surefire-booter:jar:3.0.0-SNAPSHOT from the shaded jar.
[INFO] Excluding org.apache.maven:maven-toolchain:jar:3.0-alpha-2 from the shaded jar.
[INFO] Excluding org.apache.maven.shared:maven-artifact-transfer:jar:0.11.0 from the shaded jar.
[INFO] Excluding org.apache.maven.shared:maven-common-artifact-filters:jar:3.1.0 from the shaded jar.
[INFO] Excluding commons-codec:commons-codec:jar:1.11 from the shaded jar.
[INFO] Excluding org.slf4j:slf4j-api:jar:1.7.5 from the shaded jar.
[INFO] Excluding org.codehaus.plexus:plexus-java:jar:1.0.3 from the shaded jar.
[INFO] Excluding org.ow2.asm:asm:jar:7.0 from the shaded jar.
[INFO] Excluding com.thoughtworks.qdox:qdox:jar:2.0-M9 from the shaded jar.
[INFO] Excluding org.apache.maven:maven-model:jar:3.0 from the shaded jar.
[INFO] Excluding org.apache.maven:maven-artifact:jar:3.0 from the shaded jar.
[INFO] Excluding org.apache.maven:maven-plugin-api:jar:3.0 from the shaded jar.
[INFO] Excluding org.sonatype.sisu:sisu-inject-plexus:jar:1.4.2 from the shaded jar.
[INFO] Excluding org.sonatype.sisu:sisu-inject-bean:jar:1.4.2 from the shaded jar.
[INFO] Excluding org.sonatype.sisu:sisu-guice:jar:noaop:2.1.7 from the shaded jar.
[INFO] Excluding org.codehaus.plexus:plexus-utils:jar:2.0.4 from the shaded jar.
[INFO] Excluding org.codehaus.plexus:plexus-classworlds:jar:2.2.3 from the shaded jar.
[INFO] Excluding org.codehaus.plexus:plexus-component-annotations:jar:1.7.1 from the shaded jar.
[INFO] Including org.apache.maven.shared:maven-shared-utils:jar:3.1.0 in the shaded jar.
[INFO] Including org.apache.commons:commons-lang3:jar:3.8.1 in the shaded jar.
[INFO] Including commons-io:commons-io:jar:2.6 in the shaded jar.
[INFO] Minimizing jar org.apache.maven.plugins:maven-failsafe-plugin:maven-plugin:3.0.0-SNAPSHOT
[INFO] Minimized 510 -> 105 (20%)
[INFO] Replacing original artifact with shaded artifact.
[INFO] Replacing C:\Users\ksuszynski\git\maven-surefire\maven-failsafe-plugin\target\maven-failsafe-plugin-3.0.0-SNAPSHOT.jar with C:\Users\ksuszynski\git\maven-surefire\maven-failsafe-plugin\target\maven-failsafe-plugin-3.0.0-SNAPSHOT-shaded.jar
[INFO] Dependency-reduced POM written at: C:\Users\ksuszynski\git\maven-surefire\maven-failsafe-plugin\dependency-reduced-pom.xml
[INFO]
[INFO] --- maven-invoker-plugin:3.2.0:install (pre-its) @ maven-failsafe-plugin ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Maven Failsafe Plugin 3.0.0-SNAPSHOT:
[INFO]
[INFO] Maven Failsafe Plugin .............................. FAILURE [ 20.944 s]
[INFO] Maven Surefire Report Plugin ....................... SKIPPED
[INFO] Maven Surefire Integration Tests ................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  21.382 s
[INFO] Finished at: 2019-05-23T11:05:28+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-invoker-plugin:3.2.0:install (pre-its) on project maven-failsafe-plugin: Failed to install project dependencies: MavenProject: org.apache.maven.plugins:maven-failsafe-plugin:3.0.0-SNAPSHOT @ C:\Users\ksuszynski\git\maven-surefire\maven-failsafe-plugin\dependency-reduced-pom.xml: Failed to install project artifacts: null: Failed to install POM: null: NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-invoker-plugin:3.2.0:install (pre-its) on project maven-failsafe-plugin: Failed to install project dependencies: MavenProject: org.apache.maven.plugins:maven-failsafe-plugin:3.0.0-SNAPSHOT @ C:\Users\ksuszynski\git\maven-surefire\maven-failsafe-plugin\dependency-reduced-pom.xml
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to install project dependencies: MavenProject: org.apache.maven.plugins:maven-failsafe-plugin:3.0.0-SNAPSHOT @ C:\Users\ksuszynski\git\maven-surefire\maven-failsafe-plugin\dependency-reduced-pom.xml
    at org.apache.maven.plugins.invoker.InstallMojo.installProjectDependencies (InstallMojo.java:468)
    at org.apache.maven.plugins.invoker.InstallMojo.execute (InstallMojo.java:184)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to install project artifacts: null
    at org.apache.maven.plugins.invoker.InstallMojo.installProjectArtifacts (InstallMojo.java:343)
    at org.apache.maven.plugins.invoker.InstallMojo.installProjectDependencies (InstallMojo.java:462)
    at org.apache.maven.plugins.invoker.InstallMojo.execute (InstallMojo.java:184)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoExecutionException: Failed to install POM: null
    at org.apache.maven.plugins.invoker.InstallMojo.installProjectPom (InstallMojo.java:401)
    at org.apache.maven.plugins.invoker.InstallMojo.installProjectArtifacts (InstallMojo.java:325)
    at org.apache.maven.plugins.invoker.InstallMojo.installProjectDependencies (InstallMojo.java:462)
    at org.apache.maven.plugins.invoker.InstallMojo.execute (InstallMojo.java:184)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
Caused by: java.lang.NullPointerException
    at org.apache.maven.plugins.invoker.InstallMojo.installProjectPom (InstallMojo.java:387)
    at org.apache.maven.plugins.invoker.InstallMojo.installProjectArtifacts (InstallMojo.java:325)
    at org.apache.maven.plugins.invoker.InstallMojo.installProjectDependencies (InstallMojo.java:462)
    at org.apache.maven.plugins.invoker.InstallMojo.execute (InstallMojo.java:184)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at sun.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:498)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)
```